### DISCUSSION TITLE
telescope picker with devicons and other improvements

### DISCUSSION THREAD

**Discussion by renerocksai** [nvim-telekasten/telekasten.nvim#10] [Announcements]
I pimped Telekasten's telescope file picker to support devicons (see screenshot below, there are little markdown icons next to the filenames).

This picker is now used for all Telescope actions in Telekasten.nvim. It differs from the default file picker:

- uses plenary.scan_dir synchronously instead of external jobs
- pre-sorts the file list in descending order (nice for dates, most recent first)
- (also supports devicons)
- displays subdirs if necessary
    - e.g. when searching for daily notes, no subdirs are displayed
    - but when entering a date in find_notes, the daily/ and weekly/ subdirs are displayed

![image](https://user-images.githubusercontent.com/30892199/143976131-f5274623-9b6f-4550-a1ea-3f3c93bf2ac1.png)

**Comment by lokesh-krishna** [comment]
Just stopping to say I love it!
